src/xpcc/driver/position/vl53l0.hpp

```diff
@@ -0,0 +1,531 @@
+	InterruptClear : uint8_t
+	{
+		Range = Bit0,
+	};
+	XPCC_FLAGS8(InterruptClear);
+
+	// GPIO__HV_MUX_ACTIVE_HIGH
+	enum class
+	GpioConfig : uint8_t
+	{
+		InterruptPolarityHigh = Bit4
+	};
+	XPCC_FLAGS8(GpioConfig);
+
+	/// RESULT__INTERRUPT_STATUS_GPIO
+	enum class
+	InterruptStatus : uint8_t
+	{
+		Disabled = 0,
+		LevelLow = 1,		///< (value < thresh_low)
+		LevelHigh = 2,		///< (value > thresh_high)
+		OutOfWindow = 3,	///< (value < thresh_low  OR  value > thresh_high)
+		NewSampleReady = 4,
+	};
+	XPCC_FLAGS8(InterruptStatus);
+	
+	/// RESULT__RANGE_STATUS
+	enum class
+	RangeStatus : uint8_t
+	{
+		// When set to 0, the device is busy
+		DeviceReady = Bit0,
+	};
+	XPCC_FLAGS8(RangeStatus);
+
+	enum class
+	RangeErrorCode : uint8_t
+	{
+		NoError = 0x0,
+		VCSEL_ContinuityTest = 0x1,
+		VCSEL_WatchdogTest = 0x2,
+		VHV_NoValueFound = 0x3,
+		MSRC_NoTarget = 0x4, 
+		SNR_Check = 0x5,
+		RangePhaseCheck = 0x6,
+		SigmaThresholdCheck = 0x7,
+		TCC = 0x8,
+		PhaseConsistency = 0x9,
+		MinClip = 0xA,
+		RangeComplete = 0xB,
+		RangingAlgoUnderflow = 0xC,
+		RangingAlgoOverflow = 0xD,
+		RangeIgnoreThreshold = 0xE,
+	};
+	typedef xpcc::Configuration<RangeStatus_t, RangeErrorCode, 0x0F, 3> RangeErrorCode_t;
+
+	/// MSRC__CONFIG_CONTROL
+	/// MSRC = "minimum signal rate check"
+	enum class
+	MSRCConfig : uint8_t
+	{
+		DisableRateCheck = Bit1,
+ 		DisablePreRangeCheck = Bit4,
+	};
+	XPCC_FLAGS8(MSRCConfig);
+
+public:
+	using Control_t = xpcc::FlagsGroup<
+			Start_t,
+			MeasurementSequenceStep_t,
+			InterruptConfig_t,
+			InterruptClear_t,
+			GpioConfig_t,
+			InterruptStatus_t,
+			RangeStatus_t,
+			MSRCConfig_t>;
+
+	static constexpr uint8_t ModelID[2] = { 0xEE, 0xAA };
+	static constexpr uint8_t RevisionID = 0x10;
+
+	// 30000 us = 30 ms
+	static constexpr uint32_t DefaultMeasurementTime = 30000;
+
+	// required signal rate per time period (unknown unit)
+	// this is the default value used by ST and will probably work fine
+	static constexpr uint16_t DefaultSignalRateLimit = 32;
+
+	struct TimeOverhead
+	{
+		// in microseconds
+		static constexpr uint16_t Start = 1910;
+		static constexpr uint16_t End = 960;
+		static constexpr uint16_t MSRC = 660;
+		static constexpr uint16_t TCC = 590;
+		static constexpr uint16_t DSS = 690;
+		static constexpr uint16_t PreRange = 660;
+		static constexpr uint16_t FinalRange = 550; 
+	};
+
+public:
+	struct xpcc_packed
+	Data
+	{
+		template< class I2cMaster >
+		friend class Vl53l0;
+
+		static constexpr uint16_t InvalidDistance = 8190;
+
+		Data() : distance{InvalidDistance} {}
+
+		/// @return the distance in millimeters
+		inline uint16_t
+		getDistance() { return distance; }
+
+		inline bool
+		isValid()
+		{
+			if((distance & InvalidDistance) == InvalidDistance) {
+				return false;
+			}
+
+			return (error == RangeErrorCode::RangeComplete
+				|| error == RangeErrorCode::NoError);
+		}
+		
+		inline RangeErrorCode
+		getRangeError() { return error; }
+	private:
+		uint16_t distance;
```

Comment:
Hm, this may break endianness (in theory).

The idea behind making a Data class and instantiating it separately is that it would allow us to place several data objects into a packed struct and just send that sequentially over a network.
This means that the endianness of the data object must the the same on all devices, and the easiest way to do that is just to keep it the way it falls out of the sensor.
I would therefore prefer, if you just stored the raw bytes in the Data class and did the endianness conversion in the `Data::getDistance()`.

But of course this is all BS, since every platform we support is little-endian, incl. every Intel OS, so this is a bit academic. I'd be fine if you left it like this.